tests/utils/testrunner.go

```diff
@@ -0,0 +1,73 @@
+// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+// ------------------------------------------------------------
+
+package utils
+
+import (
+	kube "github.com/dapr/dapr/tests/platforms/kubernetes"
+)
+
+// runnerFailExitCode is the exit code when test runner setup is failed
+const runnerFailExitCode = 1
+
+// testingMInterface interface is used for testing TestRunner
+type testingMInterface interface {
+	Run() int
+}
+
+// testingPlatform defines the testing platform for test runner
+type testingPlatform interface {
+	setup() error
+	tearDown() error
+
+	AcquireAppExternalURL(name string) string
+	AddTestApps(apps []kube.AppDescription) error
+	InstallApps() error
+}
+
+// TestRunner holds appmanager
+type TestRunner struct {
+	id string
+	// TODO: Needs to define kube.AppDescription more general struct for Dapr app
+	initialApps []kube.AppDescription
+	Platform    testingPlatform
```

Comment:
Why to have Platform public?

Reply:
This is intentionally exposed to public. For example, in order to get external url for the specific app, we have to add this method to AppManager in platform/kubernetes and add this method to KubePlatform in utils now.

If we make `Platform` private, then we have to expose another method to call KubePlatform.AcquireAppExternalURL, which is redundant. That's why I made it public.